run-tests.php

```diff
@@ -2959,8 +2964,19 @@ function generate_array_diff(array $ar1, array $ar2, bool $is_reg, array $w): ar
     return $diff;
 }
 
+function generate_diff_external(string $diff_cmd, string $exp_file, string $output_file): string
+{
+    return shell_exec("{$diff_cmd} $exp_file $output_file");
+}
+
 function generate_diff(string $wanted, ?string $wanted_re, string $output): string
 {
+    global $environment;
```

Comment:
This block is no longer necessary since you're now calling `generate_diff_external` directly.